interface/src/app/node/node.spec.js

```diff
@@ -0,0 +1,246 @@
+      mockGenes[1].gene.standard_name);
+    expect(elementScope.genes[2].desc).toEqual(
+      mockGenes[2].gene.description);
+    expect(elementScope.hypoPercentage).toEqual(33);
+
+    // Confirm that the rendered HTML contains data in mockGenes too:
+    var tdElements = element.find('td');
+    expect(tdElements.text()).toContain(mockGenes[0].gene.standard_name);
+    expect(tdElements.text()).toContain(mockGenes[1].gene.description);
+    expect(tdElements.text()).toContain(mockGenes[2].gene.systematic_name);
+  });
+
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+});
+
+
+describe('<high-range-exp> directive', function() {
+  beforeEach(module('adage.node'));
+
+  beforeEach(module('node/high_range_exp.tpl.html'));
+
+  var $compile, $httpBackend, $rootScope;
+  beforeEach(inject(function(_$compile_, _$httpBackend_, _$rootScope_) {
+    $compile = _$compile_;
+    $httpBackend = _$httpBackend_;
+    $rootScope = _$rootScope_;
+  }));
+
+  it('should render HTML correctly', function() {
+    var parentScope = $rootScope.$new();
+    parentScope.nodeID = 123;
+    parentScope.topExp = 20;
+    var testHTML = '<high-range-exp node-id="{{nodeID}}" ' +
+        ' top-exp="{{topExp}}"></high-range-exp>';
+
+
+    var uri1 = '/api/v0/activity/?limit=0&node=' + parentScope.nodeID;
+    var uri2 = '/api/v0/experiment/?limit=0&node=' + parentScope.nodeID;
+
+    var element = $compile(testHTML)(parentScope);
+
+    // Mocked activity response data:
+    var mockActivity = [
+      {id: 1, node: 123, sample: 1001, value: 0.0854334209211516}
+    ];
+
+    // Mocked experiment response data:
+    var mockExperiment = [{
+      'accession': 'E-GEOD-00001',
+      'description': 'Sample description',
+      'name': 'Sample experiment name',
+      'resource_uri': '/api/v0/experiment/E-GEOD-00001/',
+      'sample_set': ['/api/v0/sample/1/', '/api/v0/sample/2/']
+    }];
+
+    $httpBackend.expectGET(uri1).respond({objects: mockActivity});
+    $httpBackend.expectGET(uri2).respond({objects: mockExperiment});
+    parentScope.$digest(); // Start Angular's digest cycle manually.
+
+    // Get isolated scope of <high-range-exp> directive:
+    var elementScope = element.isolateScope();
+
+    // Confirm that the queryStatus is 'Connecting to the server ...' if
+    // server has not returned a response.
+    expect(elementScope.queryStatus).toBe('Connecting to the server ...');
+    // Confirm the text in rendered html too:
+    var divElement = element.find('div');
+    expect(divElement.text()).toBe('Connecting to the server ...');
+
+
+    // Make backend data available to the directive.
+    $httpBackend.flush();
+
+    // Confirm that properties in directive are what we expect them to be
+    // The queryStatus should now be an empty string.
+    expect(elementScope.queryStatus).toBe('');
+  });
+
+  it('should get the appropriate activity values', function() {
```

Comment:
I noticed that a few variables (`uri1`, `uri2`, `testHTML`, `element`, `elementScope`, etc) in this test case are set to same values as those in the first case (line 102). You can save a few lines by setting them once in `beforeEach()`. If you plan to keep the two cases completely independent from each other, then you can leave the code as is.